UI/MainScreen.cpp

```diff
@@ -534,7 +538,23 @@ UI::EventReturn GameBrowser::LastClick(UI::EventParams &e) {
 }
 
 UI::EventReturn GameBrowser::BrowseClick(UI::EventParams &e) {
+#if PPSSPP_PLATFORM(IOS) || PPSSPP_PLATFORM(MAC)
+	DarwinDirectoryPanelCallback callback = [this] (Path thePathChosen) {
+		File::FileInfo info;
+		if (!File::GetFileInfo(thePathChosen, &info))
+			return;
+		
+		if (info.isDirectory)
+			SetPath(thePathChosen); // user selected a dir, go there
+		else
+			LaunchFile(screenManager_, thePathChosen); // it's a file, try open it
+	};
+	
+	DarwinFileSystemServices services;
+	services.presentDirectoryPanel(callback, /* allowFiles = */ true, /* allowDirectorites = */ true);
+#else
```

Comment:
I would prefer if we didn't have to call this from here directly, and instead could do this from the System_SendMessage handler, the "browse_folder" case. Like for Windows in https://github.com/hrydgard/ppsspp/blob/master/Windows/main.cpp#L397 .

Reply:
> I would prefer if we didn't have to call this from here directly, and instead could do this from the System_SendMessage handler, the "browse_folder" case. Like for Windows in https://github.com/hrydgard/ppsspp/blob/master/Windows/main.cpp#L397 .

What about getting the needed `screenManager`? And how would we call `SetPath` / `LaunchFile`?

Reply:
You'd send "browse_fileSelect" or "browse_folderSelect" messages from the callback, then those would get handled by the existing handlers that MainScreen already has.